docs/en/writing-running-appium/android/espresso-datamatcher-selector.md

```diff
@@ -0,0 +1,338 @@
+    }
+}, {
+    contentDescription = Seek Bar,
+    title = Seek Bar,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.SeekBar1
+    }
+}, {
+    contentDescription = Spinner,
+    title = Spinner,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.Spinner1
+    }
+}, {
+    contentDescription = Splitting Touches across Views,
+    title = Splitting Touches across Views,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.SplitTouchView
+    }
+}, {
+    contentDescription = Switches,
+    title = Switches,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.Switches
+    }
+}, {
+    contentDescription = System UI Visibility,
+    title = System UI Visibility,
+    intent = Intent {
+        cmp = io.appium.android.apis / .ApiDemos(has extras)
+    }
+}, {
+    contentDescription = Tabs,
+    title = Tabs,
+    intent = Intent {
+        cmp = io.appium.android.apis / .ApiDemos(has extras)
+    }
+}, {
+    contentDescription = TextClock,
+    title = TextClock,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.TextClockDemo
+    }
+}, {
+    contentDescription = TextFields,
+    title = TextFields,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.TextFields
+    }
+}, {
+    contentDescription = TextSwitcher,
+    title = TextSwitcher,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.TextSwitcher1
+    }
+}, {
+    contentDescription = Visibility,
+    title = Visibility,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.Visibility1
+    }
+}, {
+    contentDescription = WebView,
+    title = WebView,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.WebView1
+    }
+}, {
+    contentDescription = WebView2,
+    title = WebView2,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.WebView2
+    }
+}, {
+    contentDescription = WebView3,
+    title = WebView3,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.WebView3
+    }
+}
+```
+
+This lists all of the adapters that "back up" the `ListView` and can be targeted using a datamatcher selector.
+
+Here's a code snippet that shows how to locate and click `TextClock`:
+
+```js
+driver.findElementById("list")
+  .findElement("-android datamatcher", JSON.stringify({
+    "name": "hasEntry",
+    "args": ["title", "TextClock"]
+  }))
+  .click();
+```
+
+This selector is equivalent to writing this matcher in Espresso
+
+```java
+onData(hasEntry("title", "textClock")
+  .inAdapterView(withId("android:id/list))
+  .perform(click());
+```
+
+In the above example, we select the parent `AdapterView` using an id selector and then select the child of that view. Locating the parent `AdapterView` is not necessary if the Activity only has one view. In that case, it can be omitted and it will default to the one view and we can write the selector like this:
```

Comment:
It might be also handy to mention the link to https://developer.android.com/training/testing/espresso/lists